### PR TITLE
[Term Entry] Neural Networks ai gradient descent

### LINKED ISSUE #4526: [Term Entry] Neural Networks ai gradient descent
### Reason/inspiration (optional)

"We would like a new term entry in the `AI` concept for [neural-network](https://www.codecademy.com/resources/docs/ai/neural-networks): gradient descent. The entry should go in a new file under `docs/content/ai/concepts/neural-networks/terms/gradient-descent/gradient-descent.md`.

The entry should describe the term and how it is used. Also include: 
- An "Example" section that explains to readers how to use concept for gradient descent.
- Should contain information for cncept and syntax.
- Should have a Python codebyte with a working code snippet showcasing this concept.
Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue."

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by PragatiVerma18**
### Description

Add a new term entry in the AI concept for [neural-network](https://www.codecademy.com/resources/docs/ai/neural-networks): gradient descent.

### Issue Solved

Closes #4526

### Type of Change

- [x] Adding a new entry
- [ ] Editing an existing entry (fixing a typo, bug, issues, etc)
- [ ] Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/ai/concepts/neural-networks/terms/gradient-descent/gradient-descent.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+- `gradient_of_cost_function`: The gradient (derivative) of the cost function with respect to the parameters.
+
+## Syntax
+
+Here's a basic syntax for Gradient Descent in the context of machine learning, specifically for updating the model parameters (weights) in order to minimize the cost function:
+
+```pseudo
+# Initialize parameters (weights) and learning rate
+theta = initial_value  # Parameters (weights)
+learning_rate = value  # Learning rate
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/gradient-descent/gradient-descent.md:51]:
```suggestion
theta = initial_value  # Model Parameters (weights)
learning_rate = value  # Learning rate (step size)
```

File: content/ai/concepts/neural-networks/terms/gradient-descent/gradient-descent.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+iterations = number_of_iterations  # Number of iterations
+
+# Repeat until convergence
+for i in range(iterations):
+    # Calculate the gradient of the cost function
+    gradient = compute_gradient(X, y, theta)
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/gradient-descent/gradient-descent.md:57]:
```suggestion
    gradient = compute_gradient(X, y, theta) # Gradient calculation
```

File: content/ai/concepts/neural-networks/terms/gradient-descent/gradient-descent.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+# Parameters initialization
+theta = 0.0  # Initial weight
+learning_rate = 0.01  # Step size
+iterations = 1000  # Number of iterations
+
+# Cost function (Mean Squared Error)
+def compute_cost(X, y, theta):
+    m = len(y)
+    cost = (1/(2*m)) * np.sum((X*theta - y)**2)
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/gradient-descent/gradient-descent.md:86]:
```suggestion
    cost = (1/(2*m)) * np.sum((X*theta - y)**2) # The cost function for linear regression
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @PragatiVerma18 , thank you for contributing to Codecademy Docs, the entry is nicely written! 😄 

I've suggested a few changes, could you please review and modify those at your earliest convenience? Thank you! 😃

**Comment by PragatiVerma18** [comment]
Hey @mamtawardhani, I am done with the changes. Please review. Thanks. 🫡

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @PragatiVerma18  😄 

The entry looks good to be merged! 🚀
